guides/common/modules/proc_using-vmware-cloud-init-and-userdata-templates-for-provisioning.adoc

```diff
@@ -0,0 +1,198 @@
+
+For security purposes, you must install a CA certificate to use HTTPs for all communication.
+This procedure includes steps to clean the virtual machine so that no unwanted information transfers to the image you use for provisioning.
+
+If you have an image with `cloud-init`, you must still follow this procedure to enable `cloud-init` to communicate with {Project} because `cloud-init` is disabled by default.
+
+ifndef::satellite[]
+These instructions are for Red Hat Enterprise Linux or Fedora OS, follow similar steps for other Linux distributions.
+endif::[]
+
+.Procedure
+. On the virtual machine that you use to create the image, install `cloud-init`, `open-vm-tools`, and `perl`:
++
+[options="nowrap" subs="+quotes"]
+----
+# yum -y install cloud-init open-vm-tools perl
+----
+. Disable network configuration by `cloud-init`:
++
+[options="nowrap" subs="+quotes"]
+----
+# cat << EOM > /etc/cloud/cloud.cfg.d/01_network.cfg
+network:
+  config: disabled
+EOM
+----
+. Configure `cloud-init` to fetch data from {Project}:
++
+[options="nowrap" subs="+attributes"]
+----
+# cat << EOM > /etc/cloud/cloud.cfg.d/10_datasource.cfg
+datasource_list: [NoCloud]
+datasource:
+  NoCloud:
+    seedfrom: https://{foreman-example-com}/userdata/
+EOM
+----
+. Configure modules to use in `cloud-init`:
++
+[options="nowrap" subs="+quotes"]
+----
+# cat << EOM > /etc/cloud/cloud.cfg
+cloud_init_modules:
+ - bootcmd
+
+cloud_config_modules:
+ - runcmd
+
+cloud_final_modules:
+ - scripts-per-once
+ - scripts-per-boot
+ - scripts-per-instance
+ - scripts-user
+ - phone-home
+
+system_info:
+  distro: rhel
+  paths:
+    cloud_dir: /var/lib/cloud
+    templates_dir: /etc/cloud/templates
+  ssh_svcname: sshd
+EOM
+----
+. Enable the CA certificates for the image:
++
+[options="nowrap" subs="+quotes"]
+----
+# update-ca-trust enable
+----
+. Download the `katello-server-ca.crt` file from {ProjectServer}:
+ifdef::foreman-el,katello[]
+You must have the Katello plugin installed to complete this step.
+For {ProjectServer} deployments, copy the CA certificate from the Apache configuration.
+endif::[]
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# wget -O /etc/pki/ca-trust/source/anchors/cloud-init-ca.crt http://_{foreman-example-com}_/pub/katello-server-ca.crt
+----
+. To update the record of certificates, enter the following command:
++
+[options="nowrap" subs="+quotes"]
+----
+# update-ca-trust extract
+----
+. Use the following commands to clean the image:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# systemctl stop rsyslog
+# systemctl stop auditd
+# package-cleanup --oldkernels --count=1
+# {package-clean} all
+----
+. Use the following commands to reduce logspace, remove old logs, and truncate logs:
++
+[options="nowrap" subs="+quotes"]
+----
+# logrotate -f /etc/logrotate.conf
+# rm -f /var/log/*-???????? /var/log/*.gz
```

Comment:
I had a comment here. I don't know where it went. 
Can you test locally? When I build, the asterisks here disappears ` /var/log/*.gz` becomes ` /var/log/.gz`

Reply:
@maximiliankolb this was the weird thing I mentioned in my overall feedback

Reply:
Fixed it.